src/ec2/sshKeyPair.ts

```diff
@@ -0,0 +1,39 @@
+/*!
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+import * as fs from 'fs-extra'
+import { ToolkitError } from '../shared/errors'
+import { ChildProcess } from '../shared/utilities/childProcess'
+
+export class SshKeyPair {
+    private publicKeyPath: string
+    private constructor(keyPath: string) {
+        this.publicKeyPath = `${keyPath}.pub`
+    }
+
+    public static async getSshKeyPair(keyPath: string) {
+        const keyExists = await fs.pathExists(keyPath)
+        if (!keyExists) {
+            await SshKeyPair.generateSshKeyPair(keyPath)
+        }
+        return new SshKeyPair(keyPath)
+    }
+
+    public static async generateSshKeyPair(keyPath: string): Promise<void> {
+        const process = new ChildProcess(`ssh-keygen`, ['-t', 'rsa', '-N', '', '-q', '-f', keyPath])
+        const result = await process.run()
+        if (result.exitCode !== 0) {
+            throw new ToolkitError('ec2: Failed to generate ssh key')
```

Comment:
Would it make sense to attach stdout to the error, assuming the error message is in stdout? Like this:
```suggestion
            throw new ToolkitError('ec2: Failed to generate ssh key', { details: { stdout: result.stdout } })
```

Reply:
What exactly does attaching stdout do? If it is attached, will it not log the error to console?

Reply:
It should still get logged. Attaching it establishes a clear relationship between the message and the output whereas right now the two things aren't explicitly connected.

Reply:
Should this be something we do on each error we throw? Or what is it about this error that encourages us to explicitly link them?